injected/src/features/api-manipulation.js

```diff
@@ -0,0 +1,118 @@
+import ContentFeature from '../content-feature';
+import { processAttr } from '../utils';
+
+export default class ApiManipulation extends ContentFeature {
+    init() {
+        const apiChanges = this.getFeatureSetting('apiChanges');
+        if (apiChanges) {
+            for (const scope in apiChanges) {
+                const change = apiChanges[scope];
+                if (!this.checkIsValidAPIChange(change)) {
+                    continue;
+                }
+                this.applyApiChange(scope, change);
+            }
+        }
+    }
+
+    /**
+     * Checks if the config API change is valid.
+     * @param {any} change
+     * @returns {change is APIChange}
+     */
+    checkIsValidAPIChange(change) {
+        if (typeof change !== 'object') {
+            return false;
+        }
+        if (change.type === 'remove') {
+            return true;
+        }
+        if (change.type === 'descriptor') {
+            if (change.enumerable && typeof change.enumerable !== 'boolean') {
+                return false;
+            }
+            if (change.configurable && typeof change.configurable !== 'boolean') {
+                return false;
+            }
+            return typeof change.getterValue !== 'undefined';
+        }
+        return false;
+    }
+
+    // TODO move this to schema definition imported from the privacy-config
+    /**
+     * @typedef {Object} APIChange
+     * @property {"remove"|"descriptor"} type
+     * @property {any} [getterValue] - The value returned from a getter.
+     * @property {boolean} [enumerable] - Whether the property is enumerable.
+     * @property {boolean} [configurable] - Whether the property is configurable.
+     */
+
+    /**
+     * Applies a change to DOM APIs.
+     * @param {string} scope
+     * @param {APIChange} change
+     * @returns {void}
+     */
+    applyApiChange(scope, change) {
+        const response = this.getGlobalObject(scope);
+        if (!response) {
+            return;
+        }
+        const [obj, key] = response;
+        if (change.type === 'remove') {
+            this.removeApiMethod(obj, key);
+        } else if (change.type === 'descriptor') {
+            this.wrapApiDescriptor(obj, key, change);
+        }
+    }
+
+    /**
+     * Removes a method from an API.
+     * @param {object} api
+     * @param {string} key
+     */
+    removeApiMethod(api, key) {
+        try {
+            if (key in api) {
```

Comment:
I think this is the safest way to check for the existence of the `key`, but happy to be wrong. Ideally we don't want to call the getter if it exists. For example `if (api[key]) {` threw.

Reply:
Supposedly we'll only want to manipulate own properties, so how about using `Object.hasOwnProperty()`?

Reply:
Isn't there a risk of that being hooked on the regular? Happy to change though.

Reply:
I mean, we are relying on captured-globals all over the place now, so this should work too.
Practically it's not a big deal because hopefully we'll never do this. And if we do, it wouldn't work anyways because the wrapper function below works only on own properties.